File: 88.merge_sorted_array/merge.js
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function(nums1, m, nums2, n) {
    var i = 0, j = 0;
    while(i < m && j < m){
        if (nums1[i] >= nums2[j]){
            nums1.splice(i,0,nums2[j]);//start, deleteCount, addItem
            j++;
        }else{
            i++;
        }
    }
    if(i === m){
        
        nums1 = nums1.concat(nums2.slice(j, n));
    }
    console.log(nums1);
};

merge([0],0,[1,2,3,4,5],5);

File: 88.merge_sorted_array/merge.py
class Solution(object):
    def merge(self, nums1, m, nums2, n):
        """
        :type nums1: List[int]
        :type m: int
        :type nums2: List[int]
        :type n: int
        :rtype: void Do not return anything, modify nums1 in-place instead.
        """
        k = m + n - 1
        # start from the end and put the larger number first
        while m > 0 and n > 0:
            if nums1[m-1] > nums2[n-1]: 
                nums1[k] = nums1[m-1]
                m -= 1
            else:
                nums1[k] = nums2[n-1]
                n -= 1
            k -= 1
        if n > 0: # if nums in nums1 have been used and nums2 has numbers left
            nums1[:n] = nums2[:n]